exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricExporter.java

```diff
@@ -0,0 +1,205 @@
+    this.metricServiceClient = client;
+    if (addUniqueIdentifier) {
+      this.uniqueIdentifier = UUID.randomUUID().toString().replaceAll("-", "").substring(0, 8);
+    }
+    this.exporterStartTime = Instant.now();
+  }
+
+  public static MetricExporter createWithDefaultConfiguration() throws IOException {
+    MetricConfiguration configuration = MetricConfiguration.builder().build();
+    return MetricExporter.createWithConfiguration(configuration);
+  }
+
+  public static MetricExporter createWithConfiguration(MetricConfiguration configuration)
+      throws IOException {
+    String projectId = configuration.getProjectId();
+    MetricServiceStub stub = configuration.getMetricServiceStub();
+
+    if (stub == null) {
+      Credentials credentials =
+          configuration.getCredentials() == null
+              ? GoogleCredentials.getApplicationDefault()
+              : configuration.getCredentials();
+
+      return MetricExporter.createWithCredentials(
+          projectId, credentials, configuration.getDeadline());
+    }
+    return MetricExporter.createWithClient(
+        projectId, MetricServiceClient.create(stub),
+        configuration.getAddUniqueIdentifier());
+  }
+
+  private static MetricExporter createWithClient(
+      String projectId,
+      MetricServiceClient metricServiceClient,
+      boolean addUniqueIdentifier) {
+    return new MetricExporter(projectId, metricServiceClient, addUniqueIdentifier);
+  }
+
+  private static MetricExporter createWithCredentials(
+      String projectId,
+      Credentials credentials,
+      Duration deadline) throws IOException {
+    MetricServiceSettings.Builder builder =
+        MetricServiceSettings.newBuilder()
+            .setCredentialsProvider(
+                FixedCredentialsProvider.create(checkNotNull(credentials, "Credentials not provided.")));
+    builder.createMetricDescriptorSettings()
+        .setSimpleTimeoutNoRetries(org.threeten.bp.Duration.ofMillis(deadline.toMillis()));
+    return new MetricExporter(projectId, MetricServiceClient.create(builder.build()), false);
+  }
+
+  @Override
+  public ResultCode export(Collection<MetricData> metrics) {
+    List<TimeSeries> allTimesSeries = new ArrayList<>();
+
+    for (MetricData metricData : metrics) {
+      MetricDescriptor descriptor = mapMetricDescriptor(metricData, uniqueIdentifier);
+      if (descriptor == null) {
+        continue;
+      }
+      metricServiceClient.createMetricDescriptor(
+          CreateMetricDescriptorRequest.newBuilder().setName(PROJECT_NAME_PREFIX + projectId)
+              .setMetricDescriptor(descriptor)
+              .build());
+
+      MetricWithLabels updateKey = new MetricWithLabels(descriptor.getType(),
+          metricData.getDescriptor().getConstantLabels());
+
+      // We are expecting one point per MetricData
+      Optional<MetricData.Point> metricPoint = metricData.getPoints().stream()
+          .reduce((first, second) -> second);
```

Comment:
Can you please explain what you're doing here? It seems that you're just getting the last element from `getPoints`? why not as
```java
metricData.getPoints().get( metricData.getPoints().size() - 1 );
```
?

Also, should we throw an error if size > 1?

Lastly, if there is exactly one data point, then we could just get the first one (and only).
Sorry if I misunderstood this code.

Reply:
Ah this was from when I didn't know that each metric data would only have one point, and used a stream because `metricData.getPoints()` returns a `Collection`. I've changed it to `metricData.getPoints().iterator().next()`, and added `logger.error()` for `size != 1`